design-toolkit/components/src/components/tree/index.tsx

```diff
@@ -0,0 +1,360 @@
+        const isVisible = Array.from(visibleKeys ?? []).includes(id);
+        const isViewable = isStatic ? true : (viewableKeys?.has(id) ?? false);
+
+        return (
+          <div
+            className={content({ variant, isViewable, isVisible })}
+            data-variant={variant}
+            data-last-of-set={isLastOfSet}
+          >
+            <VisibilityToggle
+              id={id}
+              isVisible={isVisible}
+              isViewable={isViewable}
+              isDisabled={isDisabled}
+              size={size}
+              onChange={onVisibilityChange}
+            />
+            {isNotRoot && <TreeLines level={level} isLastOfSet={isLastOfSet} />}
+            <ExpandToggle
+              hasChildItems={hasChildItems}
+              isVisible={isVisible}
+              isViewable={isViewable}
+              isExpanded={isExpanded}
+              size={size}
+              isDisabled={isDisabled}
+            />
+            <div className={display({ variant })}>
+              {typeof children === 'function'
+                ? children({
+                    ...renderProps,
+                    variant,
+                    isVisible,
+                    isViewable,
+                    defaultChildren: null,
+                  })
+                : children}
+            </div>
+            {shouldShowSelection && (
+              <SelectionToggle
+                isSelected={isSelected}
+                isDisabled={isDisabled}
+                slot='selection'
+              />
+            )}
+            {allowsDragging && (
+              <Button
+                slot='drag'
+                variant='icon'
+                size={size}
+                className={drag({ isVisible, isViewable })}
+              >
+                <Icon>
+                  <DragVert />
+                </Icon>
+              </Button>
+            )}
+          </div>
+        );
+      }}
+    </AriaTreeItemContent>
+  );
+}
+ItemContent.displayName = 'Tree.Item.Content';
+
+function ItemLabel({ children, className }: ItemTextProps) {
+  return <AriaText className={label({ className })}>{children}</AriaText>;
+}
+ItemLabel.displayName = 'Tree.Item.Label';
+
+function ItemDescription({ children, className }: ItemTextProps) {
+  const context = useContext(TreeContext);
+
+  const variant =
+    (isSlottedContextValue(context) ? undefined : context?.variant) ??
+    TreeStylesDefaults.variant;
+
+  return (
+    <AriaText
+      data-slot='description'
+      className={description({ className, variant })}
+    >
+      {children}
+    </AriaText>
+  );
+}
+ItemDescription.displayName = 'Tree.Item.Description';
+
+function ItemIcon({ children }: ItemTextProps) {
+  const context = useContext(TreeContext);
+
+  const variant =
+    (isSlottedContextValue(context) ? undefined : context?.variant) ??
+    TreeStylesDefaults.variant;
+
+  return (
+    <Icon size={variant === 'cozy' ? 'medium' : 'small'} className={icon()}>
```

Comment:
nit: There's two instances of this translation. If there are more, consider a translation file for the different variations: 

```ts
const variants = ['cozy', 'compact', 'crammed'] as const;
type VariantType = typeof variants[number];
type VariantConfig = {
  IconSize: 'medium' | 'small' | null
}

const Variations: Record<VariantType, VariantConfig> = {
    cozy: {
        IconSize: 'medium',
    },
    compact: {
        IconSize: 'small',
    },
    crammed: {
        IconSize: null, 
    }
}

```